numba/cuda/tests/cudadrv/test_managed_alloc.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+import platform
+from numba.cuda.cudadrv import driver
+from numba import cuda
+from numba.cuda.testing import unittest, ContextResettingTestCase
+from numba.cuda.testing import skip_on_cudasim
+
+
+@skip_on_cudasim('CUDA Driver API unsupported in the simulator')
+class TestManagedAlloc(ContextResettingTestCase):
+    def test_managed_alloc_driver(self):
+        # Verify that we can allocate and operate on managed
+        # memory through the CUDA driver interface.
+
+        ctx = cuda.current_context()
+
+        # CUDA Unified Memory comes in two flavors. For GPUs in the
+        # NVIDIA Kepler and Maxwell generations, managed memory
+        # allocations work as opaque, contiguous segments that can
+        # either be on the device or the host. For GPUs in the Pascal
+        # or later generations, managed memory operates on a per-page
+        # basis, so we can have arrays larger than GPU memory, where
+        # only part of them is resident on the device at one time. To
+        # ensure that this test works correctly on all supported GPUs,
+        # we'll select the size of our memory such that we only
+        # oversubscribe the GPU memory if we're on a Pascal or newer GPU
+        # (compute capability at least 6.0).
+
+        compute_capability_major = ctx.device.compute_capability[0]
+
+        # This test is unsupported on GPUs prior to the Kepler generation.
+
+        self.assertGreaterEqual(compute_capability_major, 3)
+
+        total_mem_size = ctx.get_memory_info().total
+
+        if compute_capability_major >= 6 and platform.system() == 'Linux':
```

Comment:
I'd prefer if we had two variants of this test - one with and one without oversubscription, with the one requiring CC 6 and Linux skipped if the system doesn't meet the requirements.

Reply:
This has been addressed; the oversubscription test is skipped in unit testing by default, but can be run manually if the platform matches.